test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+    await node.mempool.reset();
+  });
+
+  it('should mine to the primary/default wallet', async () => {
+    const height = 20;
+
+    for (let i = 0; i < height; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    const info = await nclient.getInfo();
+    assert.equal(info.chain.height, height);
+  });
+
+  it('should mine to the secondary/default wallet', async () => {
+    const {address} = await wallet2.createAddress('default');
+    for (let i = 0; i < 5; i++)
+      await nclient.execute('generatetoaddress', [1, address]);
+
+    assert.ok(true);
+  });
+
+  it('should have no name state indexed', async () => {
+    const names = await wallet.getNames();
+
+    assert.equal(names.length, 0);
+  });
+
+  it('should create an open and broadcast the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    assert.equal(entered, true);
+    const mempool = await nclient.getMempool();
+
+    assert.ok(mempool.includes(json.hash));
+
+    const mtx = MTX.fromJSON(json);
+
+    const opens = mtx.outputs.filter(output => output.covenant.type === types.OPEN);
+    assert.equal(opens.length, 1);
+  });
+
+  it('should create an open and not broadcast the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name,
+      broadcast: false
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    // tx is not in the mempool
+    assert.equal(entered, false);
+
+    const mtx = MTX.fromJSON(json);
+    assert.ok(mtx.hasWitness());
+
+    const tx = mtx.toTX();
+
+    // tx is valid
+    assert.ok(tx.verify(mtx.view));
+
+    const opens = tx.outputs.filter(output => output.covenant.type === types.OPEN);
+    assert.equal(opens.length, 1);
+  });
+
+  it('should create an open and not sign the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name,
+      broadcast: false,
+      sign: false
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    assert.equal(entered, false);
+
+    const mtx = MTX.fromJSON(json);
+
+    assert.equal(mtx.verify(), false);
+  });
+
+  it('should throw error with incompatible broadcast and sign options', async () => {
+    const fn = async () => await (wallet.createOpen({
+      name: name,
+      broadcast: true,
+      sign: false
+    }));
+
+    assert.rejects(fn, 'Must sign when broadcasting');
+  });
+
+  it('should fail to create open for account with no monies', async () => {
+    const info = await wallet.getAccount(accountTwo);
+    assert.equal(info.balance.tx, 0);
+    assert.equal(info.balance.coin, 0);
+
+    const fn = async () => (await wallet.createOpen({
+      name: name,
+      account: accountTwo
+    }));
+
+    assert.rejects(fn);
```

Comment:
Do you want to specify the error here?

Reply:
Should this be resolved? It is associated with an outdated commit.